go/libkb/merkle_client.go

```diff
@@ -238,17 +238,26 @@ type MerkleUserLeaf struct {
 	eldest    keybase1.KID // may be empty
 }
 
+type MerkleTeamLeaf struct {
+	TeamID  keybase1.TeamID
+	Public  *MerkleTriple
+	Private *MerkleTriple
+}
+
 type PathSteps []*PathStep
 
 type VerificationPath struct {
 	Contextified
-	uid           keybase1.UID
-	root          *MerkleRoot
-	path          PathSteps
-	uidPath       PathSteps
-	idVersion     int64
-	username      string
-	usernameCased string
+	// only filled in a user response, not filled for teams
+	userInfo struct {
+		uid           keybase1.UID
+		uidPath       PathSteps
+		idVersion     int64
+		username      string
+		usernameCased string
+	}
+	root *MerkleRoot
+	path PathSteps
 }
```

Comment:
I pulled `merkleUserInfoT` off and pass it through return values on the user codepaths. I'm surprised it works, I guess it's probably cleaner.